android/titanium/src/java/org/appcelerator/titanium/proxy/ServiceProxy.java

```diff
@@ -107,15 +116,55 @@ public void stop()
 	{
 		Log.d(TAG, "Stopping service", Log.DEBUG_MODE);
 		if (!forBoundServices) {
-			Log.d(TAG, "stop via stopService", Log.DEBUG_MODE);
-			service.stopSelf();
+			if (service != null) {
+				Log.d(TAG, "stop via stopService", Log.DEBUG_MODE);
+				service.stopSelf();
+			}
 		} else {
 			unbindService();
 		}
 	}
 
+	@Kroll.method
+	public void foregroundNotify(int notificationId, KrollProxy notificationProxy)
+	{
+		// Validate arguments.
+		if (notificationId == 0) {
+			throw new RuntimeException("Notification ID argument cannot be set to zero.");
+		}
+		if (notificationProxy == null) {
+			throw new RuntimeException("Notification object argument cannot be null.");
```

Comment:
This seems to be thrown on Sony devices, while it works for others. See the latest comment in the JIRA for more logs. It may be a dev-error, but lets be sure to validate it.

Reply:
Maybe the `Ti.App.Android.launchIntent` is `null` on Sony devices?

https://github.com/appcelerator/titanium_mobile/blob/master/android/modules/android/src/java/ti/modules/titanium/android/PendingIntentProxy.java#L58

Reply:
Yeah. It does sound like `launchIntent` is `null`. That should never happen. I'll look into it.

Reply:
I've concluded that this is not an issue with this PR. A `launchIntent` can in fact return `null` in rare cases. I've modified the test code to handle this case.